ouroboros-consensus/test-consensus/Test/Dynamic/Util/Expectations.hs

```diff
@@ -0,0 +1,71 @@
+{-# LANGUAGE NamedFieldPuns #-}
+
+module Test.Dynamic.Util.Expectations
+    ( NumBlocks (..)
+    , determineForkLength
+    ) where
+
+import           Data.Foldable (foldl')
+import qualified Data.Map.Strict as Map
+import           Data.Word (Word64)
+
+import           Ouroboros.Consensus.Protocol.Abstract (SecurityParam (..))
+import           Ouroboros.Consensus.Protocol.LeaderSchedule
+                     (LeaderSchedule (..))
+
+newtype NumBlocks = NumBlocks Word64
+
+-- | Internal accumulator of 'determineForkLength'
+data Acc = Acc
+  { maxForkLength  :: !Word64
+    -- ^ Upper bound on length of the longest fork in the network, excluding
+    -- the common prefix
+    --
+    -- Note that @0@ corresponds to /consensus/.
+  }
+
+-- | Compute a bound for the length of the final forks
+--
+-- At the end of the test, the nodes' final chains will share a common prefix
+-- (\"intersection\"), though it may be empty. Each current chain is a fork off
+-- that prefix. No fork will be longer than the result of this function.
+determineForkLength ::
+     SecurityParam
+  -> LeaderSchedule
```

Comment:
Oh, it would be nice to have a property about `roundRobinLeaderSchedule` + `determineForkLength`, just as a sanity check.

Reply:
Done